examples/notebooks/beam-ml/rag_usecase/opensearch_connector.py

```diff
@@ -0,0 +1,372 @@
+                 password: Optional[str] = os.getenv("OPENSEARCH_PASSWORD"),
+                 batch_size: int = 100
+                 ):
+        """
+
+        Args:
+        host (str): The opensearch host
+        port (int): The opensearch port
+        username (str): username of OpenSearch DB
+        password (str): password of OpenSearch DB
+        batch_size(int): Number of key, values pairs to write at once
+
+        Returns:
+        :class:`~apache_beam.transforms.ptransform.PTransform`
+
+        """
+
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self._batch_size = batch_size
+
+    def expand(self, pcoll):
+        return pcoll \
+               | "Reshuffle for Opensearch Insert" >> Reshuffle() \
+               | "Insert document into Opensearch" >> beam.ParDo(_InsertDocOpenSearchFn(self.host,
+                                                                                        self.port,
+                                                                                        self.username,
+                                                                                        self.password,
+                                                                                        self._batch_size)
+                                                                 )
+
+
+class _InsertDocOpenSearchFn(DoFn):
+    """Abstract class that takes in Opensearch
+    credentials to connect to Opensearch DB
+    """
+
+    def __init__(self,
+                 host: str,
+                 port: int,
+                 username: str,
+                 password: str,
+                 batch_size: int = 100
+                 ):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.batch_size = batch_size
+
+        self.batch_counter = 0
+        self.batch = list()
+
+        self.text_col = None
+
+    def finish_bundle(self):
+        self._flush()
+
+    def process(self, element, *args, **kwargs):
+        self.batch.append(element)
+        self.batch_counter += 1
+        if self.batch_counter >= self.batch_size:
+            self._flush()
+        yield element
+
+    def _flush(self):
+        if self.batch_counter == 0:
+            return
+
+        with _InsertDocOpenSearchSink(self.host, self.port, self.username, self.password) as sink:
+            sink.write(self.batch)
+            self.batch_counter = 0
+            self.batch = list()
+
+
+class _InsertDocOpenSearchSink(object):
+    """Class where we create Opensearch client
+    and write insertion logic in Opensearch
+    """
+
+    def __init__(self,
+                 host: str,
+                 port: int,
+                 username: str = os.getenv("OPENSEARCH_USERNAME"),
+                 password: str = os.getenv("OPENSEARCH_PASSWORD")
```

Comment:
Same input note here, but now you're requiring a value to be passed in. Not setting a default value makes more sense in the internal object since you're requiring a value to be passed up in `_InsertDocOpenSearchFn` anyway

Reply:
Thats a good point! Fixed.